### ISSUE TITLE
Rule `missing_docs` not triggered on enum/struct/class conforming a protocol

### ISSUE THREAD

**Issue by umbertovolta** [realm/SwiftLint#3663]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The rule `missing_docs` is not triggered on enums, structs, and classes that conform to a protocol (or extend another class).

##### Complete output when running SwiftLint, including the stack trace and command used

Note that the following output comes from linting the sample pasted below.

```bash
$ swiftlint lint swiftlint_missing_docs.swift 
Linting Swift files at paths swiftlint_missing_docs.swift
Linting 'swiftlint_missing_docs.swift' (1/1)
/Users/umv/Desktop/swiftlint_missing_docs.swift:4:8: warning: Missing Docs Violation: public declarations should be documented. (missing_docs)
/Users/umv/Desktop/swiftlint_missing_docs.swift:3:8: warning: Missing Docs Violation: public declarations should be documented. (missing_docs)
/Users/umv/Desktop/swiftlint_missing_docs.swift:16:10: warning: Missing Docs Violation: public declarations should be documented. (missing_docs)
/Users/umv/Desktop/swiftlint_missing_docs.swift:15:8: warning: Missing Docs Violation: public declarations should be documented. (missing_docs)
Done linting! Found 4 violations, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?

0.43.1

* Installation method used (Homebrew, CocoaPods, building from source, etc)?
 
Issue reproduced both from Homebrew and CocoaPods.

* Paste your configuration file:

```yml
only_rules:
  - missing_docs
```

* Which Xcode version are you using (check `xcodebuild -version`)?

12.5

* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
import Foundation

public enum Planet { // missing_docs triggered
  case earth // missing_docs triggered
}

public enum Color: Equatable { // nothing triggered
  case yellow // nothing triggered
}

public struct Square: MyProtocol { // nothing triggered
  public let side: Double // nothing triggered
}

public class Person { // missing_docs triggered
  public let name: String = "Clark" // missing_docs triggered
}

public class Hero: Person { // nothing triggered
  public let power: String = "Flight" // nothing triggered
}
```

**Comment by marcelofabri** [comment]
This seems to be a conscious choice from when the rule was implemented: https://github.com/realm/SwiftLint/blob/master/Source/SwiftLintFramework/Rules/Lint/MissingDocsRule.swift#L7. 

My guess is that this was done to avoid false positives for types conforming to protocols which already have documentation. This is a tricky one - maybe we should make this a configuration?

**Comment by umbertovolta** [comment]
> This seems to be a conscious choice from when the rule was implemented: [https://github.com/realm/SwiftLint/blob/master/Source/SwiftLintFramework/Rules/Lint/MissingDocsRule.swift#L7](https://github.com/realm/SwiftLint/blob/master/Source/SwiftLintFramework/Rules/Lint/MissingDocsRule.swift?rgh-link-date=2021-06-24T18%3A38%3A37Z#L7).
> 
> My guess is that this was done to avoid false positives for types conforming to protocols which already have documentation. This is a tricky one - maybe we should make this a configuration?

I see your point and it makes sense, but only for inherited property and methods. 

Personally, I prefer to add a brief comment referring to the protocol/superclass implementation (or even disable the rule for those cases), rather than risking forgetting documentation for the new fields.

So, I agree on having this configurable. 👍🏻

**Comment by SimplyDanny** [comment]
This is now optional yet still enabled by default. Setting `excludes_inherited_types` to `false` makes the rule run on inheriting types as well.
